### PR TITLE
chore: exclude actions from changes to docs

### PR DESCRIPTION
**Pull request by raisedadead**
This should prevent actions from getting triggered on changes to docs, and save a few runs.

### CODE REVIEW

**Comment by raisedadead** [comment]
The `pull_request_target` conditions don't seem to work when a PR is made from the org repo. That's fine, we can test it shortly after landing from a fork.

Couldn't exclude the Node.js workflows because they are a PR merge requirement. 

However the below is much nicer and faster:
image

**Comment by raisedadead** [comment]
Couldn't exclude the Node.js workflows because they are a PR merge requirement. But other than that would love to sanity check this. This is a nice to have only, and if it takes more effort then we should just close this without merging.

**Review by naomi-lgbt — DISMISSED** [review]
This should be good - we might have some other areas of the codebase we can safely ignore, too.

**Comment by raisedadead** [comment]
The cypress tests should have fired for changes to workflows, i.e a change to `.github`, no? I am missing something.

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 name: Cypress
 on:
   push:
+    paths:
+      - '!docs/**'
     branches-ignore:
       - 'renovate/**'
   pull_request:
+    paths:
+      - '!docs/**'
```
Comment (ojeytonwilliams) [inline .github/workflows/cypress.yml:10]:
```suggestion
    paths-ignore:
      - 'docs/**'
```
This, maybe?

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -1,9 +1,13 @@
 name: Cypress
 on:
   push:
+    paths:
+      - '!docs/**'
```
Comment (ojeytonwilliams) [inline .github/workflows/cypress.yml:5]:
```suggestion
    paths-ignore:
      - 'docs/**'
```

**Comment by ojeytonwilliams** [comment]
@raisedadead I think if you just add `!thing` to the `paths` property that means "ignore all paths matching `thing`", but if you want it to do anything you have to have include some paths explicitly:

```yaml
on:
  push:
    paths:
      - '**.js'
      - '!**.test.js'
```

Would include all .js files that aren't also .test.js files

```yaml
on:
  push:
    paths:
      - '!**.test.js'
```
wouldn't include any .js files in the first place.

File: .github/workflows/codesee-diagram.yml [hunk 3]
```diff
@@ -1,8 +1,13 @@
 on:
   push:
+    paths:
+      - '!docs/**'
```
Comment (raisedadead) [inline .github/workflows/codesee-diagram.yml:4]:
```suggestion
    paths-ignore:
      - 'docs/**'
```

File: .github/workflows/codesee-diagram.yml [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
     branches:
       - main
+
   pull_request_target:
+    paths:
+      - '!docs/**'
```
Comment (raisedadead) [inline .github/workflows/codesee-diagram.yml:10]:
```suggestion
    paths-ignore:
      - 'docs/**'
```

File: .github/workflows/codeql-analysis.yml [hunk 5]
```diff
@@ -3,8 +3,12 @@ name: 'CodeQL'
 on:
   push:
     branches: [main]
+    paths:
+      - '!docs/**'
```
Comment (raisedadead) [inline .github/workflows/codeql-analysis.yml:7]:
```suggestion
    paths-ignore:
      - 'docs/**'
```

File: .github/workflows/codeql-analysis.yml [hunk 6]
```diff
@@ -3,8 +3,12 @@ name: 'CodeQL'
...
   pull_request:
     branches: [main]
+    paths:
+      - '!docs/**'
```
Comment (raisedadead) [inline .github/workflows/codeql-analysis.yml:11]:
```suggestion
    paths-ignore:
      - 'docs/**'
```

**Comment by raisedadead** [comment]
Thanks @ojeytonwilliams that makes sense.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by raisedadead** [comment]
Time to give this a try.

**Pull request merged**